### ISSUE TITLE
cosmiconfig requires at least version 4 of Node, please upgrade :'4' < '8.99'   true   '4' < '10.xx'  false

### ISSUE THREAD

**Issue by jimwmg** [prettier/prettier#5745]
```javascript
var index$48 = function (pkg) {
  var requiredVersion = pkg.engines.node.replace(">=", ""); // '4'
  var currentVersion = process.version.replace("v", ""); // '10.x'  '11'
  if (currentVersion < requiredVersion) {
    console.error(
      "%s requires at least version %s of Node, please upgrade",
      pkg.name,
      requiredVersion
    );
    process.exit(1);
  }
};
index$48()
```

**Comment by jimwmg** [comment]
源码有问题

**Comment by j-f1** [comment]
Which version of Prettier are you running?

**Comment by Ardeshir81** [comment]
Hello! I have this issue & I have `^1.7.0` in my `package.json` which is resolved to:
```
"prettier": {
      "version": "1.7.0",
      "resolved": "https://registry.npmjs.org/prettier/-/prettier-1.7.0.tgz",
      "integrity": "sha512-kIbA3UE9sYGiVCxlWg92EOHWZqte6EAkC7DS5je6NaL8g3Uw1rWe0eH+UX4Hy5OEiR9aql2vVMHeg6lR4YGxYg==",
      "dev": true
    },
```

> `package-lock.json`

and when I search in prettier's package source code, I see: `"please-upgrade-node":"^3.0.1"`

**Comment by Ardeshir81** [comment]
Which is actually seems to be solved in prettier@1.7.1 and I only needed to `npm i -D prettier@1.7.1`.

But how are everyone expect to solve this? Like me step through `CHANGELOG.md`?
Isn't there a service like `npm audit` but for this kind of things.
like **found 1 error in your build system (1 awful, 0 moderate, 0 low). run `npm wtf fix` to fix them, or `npm wtf` for details.** and when I run `npm wtf` I get:

> Hey, your build system depends on prettier@1.7.0 which depends on an awful cosmiconfig/please-upgrade-node version which breaks your build system. upgrade to (at least) prettier@1.7.1 for fix and also consider these breaking changes:

- <considerations of breakingChange 1>
- <considerations of breakingChange 2>
- <considerations of breakingChange 3>
- <considerations of breakingChange 4>
